packages/api-client/src/components/SideNav/SideNavLink.vue

```diff
@@ -1,36 +1,34 @@
 <script setup lang="ts">
-import type { ROUTES } from '@/constants'
-import { useWorkspace } from '@/store'
-import { ScalarIcon } from '@scalar/components'
-
-type IconProps = InstanceType<typeof ScalarIcon>['$props']
+import { useLayout } from '@/hooks'
+import { type Icon, ScalarIcon } from '@scalar/components'
+import type { Component } from 'vue'
 
 defineProps<{
-  icon: IconProps['icon']
-  name: (typeof ROUTES)[number]['name'] | 'settings'
+  is?: Component | string
   active?: boolean
+  icon: Icon
 }>()
 
-const { activeWorkspace } = useWorkspace()
+const { layout } = useLayout()
 </script>
 <template>
-  <router-link
-    activeClass="active-link"
-    class="flex flex-col items-center gap-1 group no-underline"
-    :to="`/workspace/${activeWorkspace.uid}/${name}`">
-    <div
-      class="min-w-[37px] max-w-[37px] group-hover:bg-b-2 active:text-c-1 flex items-center justify-center rounded-lg p-[8px] scalar-app-nav-padding text-c-3"
-      :class="{
-        'bg-b-2 transition-none group-hover:cursor-auto !text-c-1': active,
-      }">
+  <component
+    :is="is ?? 'a'"
+    class="hover:bg-b-3 no-underline min-w-[37px] max-w-[37px] flex items-center justify-center rounded-lg p-2"
+    :class="{
+      'bg-b-3 transition-none hover:cursor-auto text-c-1': active,
```

Comment:
should we keep it `bg-b-2` on darkmode?

Reply:
never a big fan of these random px values, _usually_ means something else is wrong! If we stick to the grid then we never need px!

Reply:
@antlio most likely needs some tweaking on dark mode, me or you can fast follow on monday or I'll update this weekend :)